rskj-core/src/main/java/co/rsk/core/TransactionListExecutor.java

```diff
@@ -65,17 +68,25 @@ public TransactionListExecutor(
         this.registerProgramResults = registerProgramResults;
         this.transactionResults = transactionResults;
         this.programTraceProcessor = programTraceProcessor;
-        this.totalFees = Coin.ZERO;
         this.totalGas = 0L;
         this.i = firstTxIndex;
+        this.totalPaidFees = totalPaidFees;
+        this.remascEnabled = remascEnabled;
     }
 
     @Override
     public Boolean call() {
         long totalGasUsed = 0;
-        Coin totalPaidFees = Coin.ZERO;
 
         for (Transaction tx : transactions) {
+
+            int numberOfTransactions = block.getTransactionsList().size();
+            boolean isRemascTransaction = tx.isRemascTransaction(this.i, numberOfTransactions);
+
+            if (this.remascEnabled && isRemascTransaction) {
+                addFeesToRemasc();
+            }
```

Comment:
A suggestion for you to consider: what if we iterate over transactions.size - 1 and then do the if REMASC check outside of the loop? This would avoid checking on each tx.